backend/clubs/serializers.py

```diff
@@ -2462,13 +2456,50 @@ def create(self, validated_data):
         return super().create(validated_data)
 
     def update(self, instance, validated_data):
-        user_field = validated_data.pop("user", None)
-        if user_field:
+        if user_field := validated_data.pop("user", None):
             username = user_field.pop("username")
             user = get_user_model().objects.get(username=username)
             instance.user = user
         return super().update(instance, validated_data)
 
+    def validate(self, data):
+        username = None
+        if user_field := data.get("user") or not self.instance:
+            username = user_field.get("username")
+            user = get_user_model().objects.filter(username=username).first()
+            if not user:
+                raise serializers.ValidationError("Please provide a valid username!")
+
+        application_pk = self.context["view"].kwargs.get("application_pk")
+        application = ClubApplication.objects.filter(pk=application_pk).first()
+
+        if not application:
+            raise serializers.ValidationError("Invalid application id!")
+
+        if (
```

Comment:
Nit: autoformatter separates this across a lot of lines...can you potentially separate out the logic for readability?